app/src/forms/proxy/controller.js

```diff
@@ -19,22 +22,53 @@ module.exports = {
       const proxyHeaderInfo = await service.readProxyHeaders(req.headers);
       // find the specified external api configuration...
       const extAPI = await service.getExternalAPI(req.headers, proxyHeaderInfo);
-      if (extAPI.code != ExternalAPIStatuses.APPROVED) {
-        throw new Problem(407, 'External API has not been approved by CHEFS.');
-      }
       // add path to endpoint url if included in headers...
       const extUrl = service.createExternalAPIUrl(req.headers, extAPI.endpointUrl);
       // build list of request headers based on configuration...
       const extHeaders = await service.createExternalAPIHeaders(extAPI, proxyHeaderInfo);
+      // check for approval before we call it..
+      if (extAPI.code != ExternalAPIStatuses.APPROVED) {
+        throw new Problem(407, 'External API has not been approved by CHEFS.');
+      }
       let axiosInstance = axios.create({
         headers: extHeaders,
       });
       // call the external api
-      const { data } = await axiosInstance.get(extUrl);
+      const { data, status } = await axiosInstance.get(extUrl).catch(function (err) {
+        let message = err.message;
+        if (err.response) {
+          // The request was made and the server responded with a status code
+          // that falls out of the range of 2xx
+          log.warn(`Error returned from the external API': ${message}`);
+        } else if (err.request) {
+          message = 'External API call made, no response received.';
+          log.warn(message);
+        } else {
+          // Something happened in setting up the request that triggered an Error
+          log.warn(`Error setting up the external API request:  ${message}`);
+        }
+        // send a bad gateway, the message should contain the real status
+        throw new Problem(502, message);
+      });
       // if all good return data
-      res.status(200).json(data);
+      res.status(status).json(data);
     } catch (error) {
-      next(error);
+      if (error instanceof ProxyServiceError) {
+        // making an assumption that the form component making this call
+        // has not been setup correctly yet.
+        // formio components will call as soon as the URL is entered while designing.
+        // calls will fire before the designer has added the headers.
+        log.warn(error.message);
+        // send an empty result back with status 299 Miscellaneous Persistent Warning
+        res.status(299).json([]);
```

Comment:
Changes look great! Just a couple of questions here:
1. Instead of the `299` status, what happens if something like `400` is returned? Does this cause warnings in the UI (we don't want that) but just wondering why we'd return a `2XX` success response when it is a bad request.
2. Would `res.sendStatus(299);` have the effect you want? The code has a few different ways to return just status codes, and I'm hoping to standardize on one of them (I don't care which one, but if they all do the same thing it's a bit less confusing than asking "what's the difference" all the time). (The `sendStatus` makes for slightly shorter unit tests.)

Reply:
1. that was more for checking logs, filtering on 4xx would include these when we probably don't want them. so up to you - make the call for the status you want returned.
2. yup, `res.sendStatus(299);` will work. i was concerned that not sending a body back would mess up the formio components - seems ok with select/dropdown.

Reply:
1. Let's go with 400, since it is a "Bad Request"? We don't worry toooo much about 400s unless we see a lot unexplained ones coming from our own frontend.
2. `sendStatus` would be great!

Reply:
done - 400 and sendStatus.